src/test/regress/expected/propagate_extension_commands.out

```diff
@@ -0,0 +1,399 @@
+ count 
+-------
+     0
+(1 row)
+
+SELECT run_command_on_workers($$SELECT count(*) FROM pg_extension WHERE extname = 'seg'$$);
+ run_command_on_workers 
+------------------------
+ (localhost,57637,t,0)
+ (localhost,57638,t,0)
+(2 rows)
+
+-- restore client_min_messages after DROP EXTENSION
+RESET client_min_messages;
+-- make sure that the extension is not avaliable anymore as a distributed object
+-- TODO:: There is a bug now
+SELECT count(*) FROM citus.pg_dist_object WHERE objid = (SELECT oid FROM pg_extension WHERE extname IN ('seg', 'isn'));
+ count 
+-------
+     0
+(1 row)
+
+CREATE SCHEMA "extension'test";
+SET search_path TO "extension'test";
+-- check restriction for sequential execution
+-- enable it and see that create command errors but continues its execution by changing citus.multi_shard_modify_mode TO 'off
+BEGIN;
+	CREATE TABLE some_random_table (a int);
+	SELECT create_distributed_table('some_random_table', 'a');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+	CREATE EXTENSION seg;
+	CREATE TABLE some_random_table_2 (a int, b seg);
+	SELECT create_distributed_table('some_random_table_2', 'a');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+ROLLBACK;
+-- show that the CREATE EXTENSION command propagated even if the transaction
+-- block is rollbacked, that's a shortcoming of dependency creation logic
+SELECT run_command_on_workers($$SELECT extversion FROM pg_extension WHERE extname = 'seg'$$);
+ run_command_on_workers  
+-------------------------
+ (localhost,57637,t,1.3)
+ (localhost,57638,t,1.3)
+(2 rows)
+
+-- drop the schema and all the objects
+SET client_min_messages TO WARNING;
+DROP SCHEMA "extension'test" CASCADE;
+-- recreate for the next tests
+CREATE SCHEMA "extension'test";
+-- use  a schema name with escape character
+SET search_path TO "extension'test";
+RESET client_min_messages;
+-- remove the node, we'll add back again
+SELECT master_remove_node('localhost', :worker_2_port);
+ master_remove_node 
+--------------------
+ 
+(1 row)
+
+-- now, create a type that depends on another type, which
+-- finally depends on an extension
+BEGIN;
+	SET citus.shard_replication_factor TO 1;
+	CREATE EXTENSION seg;
+	CREATE EXTENSION isn;
+	CREATE TYPE test_type AS (a int, b seg); 
+	CREATE TYPE test_type_2 AS (a int, b test_type); 
+	CREATE TABLE t2 (a int, b test_type_2, c issn);
+	SELECT create_distributed_table('t2', 'a');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+	
+	CREATE TYPE test_type_3 AS (a int, b test_type, c issn); 
+	CREATE TABLE t3 (a int, b test_type_3);
+	SELECT create_reference_table('t3');
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+COMMIT;
+-- add the node back
+SELECT master_add_node('localhost', :worker_2_port);
+NOTICE:  Replicating reference table "t3" to the node localhost:57638
+ master_add_node 
+-----------------
```

Comment:
master_add_node()'s output could change if someone adds a node before this test. so could you do something like

SELECT 1 FROM master_add_node()

please check all the tests in the file